.github/workflows/docker-image.yml

```diff
@@ -0,0 +1,38 @@
+name: ci hive docker image
+
+on:
+  push:
```

Comment:
Some thoughts on the frequency:
  1. We would better to trigger the action for every new release.
  2. For master branch, I think we can update the image every three months.

Reply:
I think we should have a -latest with the GA version.
Also we could have a daily release about the  -dev version (or tags)
for every commit would be a bit overused.

Reply:
I think we should have 2 workflows:
1. GA workflow  - Frequency : Once per release
2. For the latest dev images.  - Frequency: Once per week?  Because on an average hive gets about 10 to 15 commits per week.(https://github.com/apache/hive/graphs/commit-activity)

This PR set up a workflow to build and publish docker images for the GA versions of hive. 
I will raise a follow-up jira to address the workflow needed for daily/dev images.

Reply:
Once per week makes sense to me for dev, if there are some limitations on the number of the dev images, for example, only keep the latest 10 images for dev.